x/swap/simulation/genesis.go

```diff
@@ -0,0 +1,85 @@
+package simulation
+
+import (
+	"fmt"
+	"math/rand"
+	"sort"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	"github.com/kava-labs/kava/x/swap/types"
+)
+
+var (
+	accs []simulation.Account
+)
+
+// GenSwapFee generates a random SwapFee in range [0.01, 1.00]
+func GenSwapFee(r *rand.Rand) sdk.Dec {
+	min := int(1)
+	max := int(100)
+	percentage := int64(r.Intn(int(max)-min) + min)
+	return sdk.NewDec(percentage).Quo(sdk.NewDec(100))
+}
+
+// GenSupportedAssets generates random allowed pools
+func GenAllowedPools(r *rand.Rand) types.AllowedPools {
+
+	numPools := (r.Intn(10) + 1)
+	pools := make(types.AllowedPools, numPools)
+	for i := 0; i < numPools; i++ {
+		tokenA, tokenB := genTokenDenoms(r)
+		for strings.Compare(tokenA, tokenB) == 0 {
+			tokenA, tokenB = genTokenDenoms(r)
+		}
+		newPool := types.NewAllowedPool(tokenA, tokenB)
+		pools[i] = newPool
+	}
+	return pools
+}
+
+func genTokenDenoms(r *rand.Rand) (string, string) {
+	tokens := []string{genTokenDenom(r), genTokenDenom(r)}
```

Comment:
should check denom[0] != denom[1]

Reply:
https://github.com/Kava-Labs/kava/pull/924/commits/8621d97b877d15a420b044b82b183a573c3226ae